test/unit/CookiesStorageTests/cookieSave_test.js

```diff
@@ -33,20 +64,125 @@ describe('The save method for cookiesStorage', () => {
   it('Saves cookies with number values', () => {
     storage.save('num', 10);
 
-    const objString = '10';
+    const numString = '10';
 
-    expect(document.cookie.includes(`ml_num=${objString}`)).toBe(true);
+    expect(document.cookie.includes(`ml_num=${numString}`)).toBe(true);
 
     removeCookie('num');
   });
 
   it('Saves cookies with boolean values', () => {
     storage.save('bool', true);
 
-    const objString = 'true';
+    const boolString = 'true';
 
-    expect(document.cookie.includes(`ml_bool=${objString}`)).toBe(true);
+    expect(document.cookie.includes(`ml_bool=${boolString}`)).toBe(true);
 
     removeCookie('bool');
   });
+
+  it('Saves multiple cookies', () => {
+    storage.save('bool', true);
+    const boolString = 'true';
+
+    storage.save('num', 10);
+    const numString = '10';
+
+    storage.save('hello', 'world');
+
+    expect(document.cookie.includes(`ml_bool=${boolString}`)).toBe(true);
+
+    expect(document.cookie.includes(`ml_num=${numString}`)).toBe(true);
+
+    expect(document.cookie.includes(`ml_hello="world"`)).toBe(true);
+
+    removeCookie('bool');
+    removeCookie('num');
+    removeCookie('hello');
+  });
+
+  it('Overwrites cookies saved with the same key', () => {
+    storage.save('hello', 'world');
+
+    expect(document.cookie.includes(`ml_hello="world"`)).toBe(true);
+
+    storage.save('hello', 'goodbye');
+
+    expect(document.cookie.includes(`ml_hello="world"`)).toBe(false);
+
+    expect(document.cookie.includes(`ml_hello="goodbye"`)).toBe(true);
+
+    removeCookie('hello');
+  });
+
+  it('Saves cookies without a prefix', () => {
+    const noPrefixStorage = new CookiesStorage({'prefix': '', 'expires': '1'});
+
+    noPrefixStorage.save('hello', 'world');
+
+    expect(document.cookie.includes(`hello="world"`)).toBe(true);
+
+    removeCookie('hello');
+  });
+
+  it('Saves cookies with a custom prefix', () => {
+    const noPrefixStorage =
+      new CookiesStorage({'prefix': 'p_', 'expires': '1'});
+
+    noPrefixStorage.save('hello', 'world');
+
+    expect(document.cookie.includes(`p_hello="world"`)).toBe(true);
+  });
+
+  it('Saves cookies with correct domain', () => {
+    const expiryInSeconds = 3;
+
+    const fullAttributeStorage = new CookiesStorage(
+      {'prefix': 'pre', 'expires': expiryInSeconds, 'domain': document.domain});
+
+    let testCookie = '';
+    spyOn(fullAttributeStorage, 'setCookie').and.callFake((cookie) => {
+      testCookie = cookie;
+    });
+
+    fullAttributeStorage.save('hello', 'goodbye');
+
+    expect(testCookie.includes(`prehello="goodbye"`)).toBe(true);
+
+    expect(testCookie.includes(`domain=${document.domain}`)).toBe(true);
+  });
+
+  it('Saves cookies with correct expiry', () => {
```

Comment:
tl;dr: I'd recommend faking out date using Jasmine's mock clock feature:
https://jasmine.github.io/api/edge/Clock.html

Long version with rationale and stuff:
This test is pretty good.  It verifies that the expiry is approximately correct (set to within 1 second of the user config), and should pass the vast majority of times it's run.  It may fail on a particularly slow machine, where the time difference between when maxExpected is recorded and when save does a similar computation is > 1 second, but runs like that will be very rare.

IMO, #1 is fine.  +/-1 second is pretty small relative to the time durations we expect to see for expirations (usually on the order of years, and probably not less than ~24 hours).

#2 is a small problem for a single test (you can rerun it a couple times to confirm that it was just a one-time flake), but gets to be a bigger problem in larger projects with many tests that have similar issues.

That said, it'd be a stronger test if it:
- Faked out the Date constructor while the code-under-test is executed.
- Had the fake date constructed always at a fixed time.
- Assert that the cookie string's expiry section is always the fixed time + the user-configured expires time.

Date is a pain to fake out by hand though.  But there's good news!  Jasmine has a handy-dandy utility for it.  They have some documentation at the link in the tl;dr, and it seems pretty easy to use.

Reply:
Thank you for the advice!